### PR TITLE
Implement ready-light relay routing

### LINKED ISSUE #719: relay: honor readiness next_action=proceed as ready_light
Parent: #718
Related: #431, #439

## Problem

`score-readiness.js` can return `next_action: "proceed"` for all-high readiness even when `bypass=false`, but `run-preflight.js` currently treats `bypass !== true` as prompt/noninteractive-fail.

That means a small, clear task without a formal Done Criteria heading can pay unnecessary relay-ready Q&A cost or fail in non-interactive mode.

## Scope

Update `skills/relay/scripts/run-preflight.js` so `buildReadinessDecision()` distinguishes:

- `ready_single`: `bypass=true`
- `ready_light`: `bypass=false` and `next_action="proceed"`
- existing prompt/noninteractive branches for unresolved readiness gaps

Do not add `not_relay_task`.

## Acceptance criteria

- `next_action="proceed"` no longer routes to prompt/noninteractive-fail solely because `bypass=false`.
- JSON output exposes an operator-readable route decision for ready-light cases.
- Existing `bypass=true` behavior remains compatible.
- Existing `qa_needed` prompt/noninteractive-fail behavior remains compatible.
- Tests cover all three cases: bypass, ready-light, qa-needed.

## Suggested tests

- Add or extend `tests/relay/scripts/run-preflight.test.js` with a fake/probe body that yields all-high readiness without a Done Criteria heading.
- Assert non-interactive ready-light proceeds instead of failing closed.

### LINKED ISSUE #720: relay-plan: consume ready_light as S-size quick planning
Parent: #718
Related: #438, #439

## Problem

`task-profile.js` already supports `size=S` and `execution_mode=quick`, but `/relay` has no clear way to pass ready-light intent into planning.

Without that connection, small tasks can still receive over-sized rubrics, extra factors, and unnecessary implementation guidance.

## Scope

Wire ready-light route intent into relay-plan's task profile and handoff guidance.

The implementation should reuse existing task-profile and rubric validation concepts rather than creating a new planning mode.

## Acceptance criteria

- Ready-light planning is represented as `size: S` and `execution_mode: quick` unless risk tags force standard/hardened behavior.
- Ready-light dispatch prompts include compact guidance: solve the observable change, avoid scope expansion, avoid unsupported defensive edge cases.
- Risk-bearing tasks cannot be downshifted to quick solely because the route was ready-light.
- Tests cover ready-light profile derivation and risk override.

## Suggested tests

- Extend `tests/relay-plan/scripts/task-profile.test.js` for ready-light intent.
- Add a risk-tag case showing `trust-boundary` or `state-machine` still routes to `fresh-context`/hardened.

### LINKED ISSUE #721: relay-plan: enforce compact rubric quality card for ready_light
Parent: #718
Related: #439

## Problem

The rubric validation docs already describe S-size mechanical tasks as 1-2 substantive factors, but this is easy to miss during planning.

Ready-light needs a visible quality card that prevents small tasks from turning into large contracts.

## Scope

Strengthen `relay-plan` validation/guidance for ready-light rubrics.

The goal is to make compactness visible and actionable before dispatch, not to add a large rubric engine.

## Acceptance criteria

- Ready-light quality card shows planning profile, prerequisites count, contract factor count, quality factor count, and dispatch action.
- Ready-light warns or blocks when substantive factors exceed 2 without explicit risk/design-bearing rationale.
- Repo-wide lint/typecheck/test remain prerequisites, not factors.
- Unsupported helper/dependency/config requirements are flagged as over-engineering risk.
- Documentation examples cover ready-light S mechanical and S design-bearing cases.

## Suggested tests

- Add reference-contract tests for ready-light quality card language.
- Add fixture rubrics showing allowed 1-factor and rejected 3-factor ready-light shapes.

### LINKED ISSUE #722: relay: route strong task-shape signals to needs_split without dispatch
Parent: #718
Related: #431, #439

## Problem

Strong task-shape signals already exist in `score-readiness.js`, but `/relay` does not expose a clear first-class route decision for oversized or semantically coupled work.

The desired behavior is not automatic decomposition. It is fail-closed shaping: stop before dispatch, propose leaves through relay-ready, and continue only after accepted handoffs exist.

## Scope

Update route preflight and `/relay` Step 1 guidance so strong task-shape signals produce `needs_split`.

`needs_split` means:

- prompt-allowed mode: invoke relay-ready proposal-first shaping
- non-interactive mode: fail closed before dispatch
- explicit bypass: allow progress but record override event

## Acceptance criteria

- Strong `task_shape` from readiness probe is visible as `route_decision=needs_split` or equivalent.
- Non-interactive `needs_split` does not dispatch.
- Prompt-allowed `needs_split` routes to relay-ready shaping, not normal planning.
- Documentation states that scripts detect shape signals only; semantic leaf boundaries require operator-approved relay-ready handoff.

## Suggested tests

- Use the existing broad product foundation fixture shape from `score-readiness.test.js`.
- Add `run-preflight` assertions for prompt-allowed and non-interactive handling.

### PR DESCRIPTION
**Pull request by sungjunlee**
## Summary
- Add relay preflight route decisions for ready_single, ready_light, readiness_prompt, and needs_split without adding a new lifecycle state.
- Preserve existing bypass and qa_needed prompt/non-interactive behavior.
- Make relay-plan treat ready_light as S-size quick planning with compact rubric guidance, while keeping risk-bearing tasks hardened.

Closes #719.
Closes #720.
Closes #721.
Closes #722.
Refs #718.

## Validation
- `node --test tests/relay-ready/scripts/*.test.js` - 64/64 pass
- `node --test tests/relay/scripts/run-preflight.test.js tests/relay-plan/scripts/task-profile.test.js tests/relay-plan/scripts/rubric-reference-contract.test.js` - 38/38 pass
- Broader combined command observed an existing `tests/relay-plan/scripts/tdd-flavor.test.js` fixture failure in `finalize-run.js` (`Unexpected end of JSON input`); reran affected ready-light/readiness suites separately as above.

## Release Notes

* **Documentation**
  * Ready-light 라우팅 결정에 대한 예제 및 선택 가이드 추가
  * 라우팅 로직 및 readiness 라벨의 의미 명확화
  * 작업 프로필 설정 가이드 개선

* **Tests**
  * Ready-light 경계 조건에 대한 테스트 케이스 추가
  * 라우팅 의사결정 로직 검증 강화

### CODE REVIEW

File: skills/relay/scripts/run-preflight.js [hunk 1]
```diff
@@ -231,32 +231,56 @@ function runReadinessProbe({ issueNumber, body, bodyFile, manifestPath }) {
   };
 }
 
+function routeDecisionFromReadiness(envelope) {
+  if (!envelope) return "readiness_prompt";
+  if (envelope.bypass === true) return "ready_single";
+  if (envelope.task_shape?.strong === true) return "needs_split";
+  if (envelope.next_action === "proceed") return "ready_light";
```
[the remark this answers is not in the document]
Comment (sungjunlee) [reply skills/relay/scripts/run-preflight.js]: **Fixed** in `cc5a858`.

```diff
- if (envelope.next_action === "proceed") return "ready_light";
+ if (envelope.next_action === "proceed" && !hasHighRiskReadinessSignal(envelope)) return "ready_light";
```

**Why:** high-risk non-bypass proceed results now stay on the existing readiness prompt / noninteractive-fail path. `probe-readiness` also emits structured `risk.high` so routing does not depend only on the bounded human summary.

**Verification:**
- `node --test tests/relay-ready/scripts/*.test.js` - 64/64 pass
- `node --test tests/relay/scripts/run-preflight.test.js tests/relay-plan/scripts/task-profile.test.js tests/relay-plan/scripts/rubric-reference-contract.test.js` - 39/39 pass
- CLI check: high-risk proceed routes to `readiness_prompt` / `noninteractive-fail`; normal proceed remains `ready_light` / `ready-light`.

File: skills/relay/scripts/run-preflight.js [hunk 2]
```diff
@@ -231,32 +231,56 @@ function runReadinessProbe({ issueNumber, body, bodyFile, manifestPath }) {
...
+  return "readiness_prompt";
```
[the remark this answers is not in the document]
Comment (sungjunlee) [reply skills/relay/scripts/run-preflight.js:239]: **Fixed** in `aa208d6`.

```diff
- if (envelope.next_action === "proceed" && !hasHighRiskReadinessSignal(envelope)) return "ready_light";
+ if (envelope.bypass === false && envelope.next_action === "proceed" && !hasHighRiskReadinessSignal(envelope)) return "ready_light";
```

**Why:** `ready_light` now requires an explicit `bypass === false`, so missing or abnormal bypass values fail closed to the existing readiness prompt / noninteractive-fail path.

**Verification:**
- Added `route decision fails closed when bypass is missing from proceed envelope`.
- `node --test tests/relay/scripts/run-preflight.test.js tests/relay-plan/scripts/task-profile.test.js tests/relay-plan/scripts/rubric-reference-contract.test.js` - 40/40 pass
- `node --test tests/relay-ready/scripts/*.test.js` - 64/64 pass

**Review by sungjunlee — COMMENTED** [review]
LGTM.

Reviewed after addressing both automated review findings:
- high-risk `next_action=proceed` no longer routes to `ready_light`; it stays on readiness prompt / noninteractive-fail.
- `ready_light` now requires explicit `bypass === false`, so incomplete envelopes fail closed.

Validation checked:
- `node --test tests/relay-ready/scripts/*.test.js` - 64/64 pass
- `node --test tests/relay/scripts/run-preflight.test.js tests/relay-plan/scripts/task-profile.test.js tests/relay-plan/scripts/rubric-reference-contract.test.js` - 40/40 pass
- GitHub Actions `test` - pass
- CodeRabbit - pass / addressed

**Comment by sungjunlee** [comment]
## Relay Review — Skipped
Reason: Manual PR: implemented directly, reviewed in PR comments, addressed Codex and CodeRabbit findings, local and GitHub checks passed.
rubric_status: unresolved-manifest
